controllers/storagecluster/backingstorageclasses.go

```diff
@@ -0,0 +1,92 @@
+package storagecluster
+
+import (
+	"fmt"
+	"reflect"
+
+	configv1 "github.com/openshift/api/config/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	v1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type backingStorageClasses struct{}
+
+// ensureCreated ensures that backing storageclasses are created for the StorageCluster
+func (obj *backingStorageClasses) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	pvReclaimPolicy := corev1.PersistentVolumeReclaimDelete
+	allowVolumeExpansion := true
+	volumeBindingMode := v1.VolumeBindingWaitForFirstConsumer
+
+	for i := range sc.Spec.BackingStorageClasses {
+		bsc := &sc.Spec.BackingStorageClasses[i]
+		if bsc.Name == "" {
+			return reconcile.Result{}, fmt.Errorf("backingStorageClass name is empty")
+		}
```

Comment:
Just mark it as failed as we need to have a storageclass name to create a storageClass.

Reply:
My issue is that a misconfigured baking storage class will block the creation of all of the other backing storageclasses. I would suggest logging an error and then skipping the problematic storageclass.

Reply:
The problem is that only a few osd might get created for the right backing storageclass. if we block it here it can be avoided. am fine doing the change please let me know your thoughts